File: Scalar Academy/4-Introduction to Problem Solving/1-Count Number Of Factor/count_number_of_factor.js
"use strict";
process.stdin.resume();
process.stdin.setEncoding("utf-8");
let inputString = "";
let currentLine = 0;
process.stdin.on("data", (inputStdin) => {
  inputString += inputStdin;
});
process.stdin.on("end", (_) => {
  inputString = inputString
    .trim()
    .split("\n")
    .map((string) => {
      return string.trim();
    });
  main();
});

function readLine() {
  return inputString[currentLine++];
}

//Above code is taking input

function main() {
  let num = parseInt(readLine());
  let factor = 0;
  for (let i = 0; i * i <= num; i++) {
    if (num % i == 0) {
      if (num / i != i) factor += 2;
      else factor += 1;
    }
  }
  console.log(factor);
}

File: Scalar Academy/4-Introduction to Problem Solving/1-Count Number Of Factor/count_number_of_factor.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    long long num;
    cin >> num;
    long long factor = 0;
    for (long long i = 1; i * i <= num; i++)
    {
        if (num % i == 0)
        {
            if (num / i != i)
                factor += 2;
            else
                factor += 1;
        }
    }
    cout << factor << endl;

    return 0;
}